application/common/components/adapters/SecureUserIdStore.php

```diff
@@ -0,0 +1,164 @@
+<?php
+namespace Sil\Idp\IdSync\common\components\adapters;
+
+use Exception;
+use GuzzleHttp\Client;
+use InvalidArgumentException;
+use Sil\Idp\IdSync\common\components\IdStoreBase;
+use Sil\Idp\IdSync\common\models\User;
+use yii\helpers\Json;
+
+class SecureUserIdStore extends IdStoreBase
+{
+    public $apiUrl = null;
+    public $apiKey = null;
+    public $apiSecret = null;
+
+    public $timeout = 45; // Timeout in seconds (per call to ID Store API).
+
+    protected $httpClient = null;
+
+    public function init()
+    {
+        $requiredProperties = [
+            'apiUrl',
+            'apiKey',
+            'apiSecret',
+        ];
+        foreach ($requiredProperties as $requiredProperty) {
+            if (empty($this->$requiredProperty)) {
```

Comment:
Is this the PHP Magic which will use the new ID_STORE_CONFIG variables?

Reply:
application/common/config/main.php line 20
```
$idStoreOptionalConfig = Env::getArrayFromPrefix('ID_STORE_CONFIG_');
```

and then lines 73-77
```
       'idStore' => ArrayHelper::merge([
            'class' => IdStoreBase::getAdapterClassFor(
                Env::get('ID_STORE_ADAPTER')
            ),
        ], $idStoreOptionalConfig),
```